docs-ref-conceptual/includes/query-azure-cli-examples.md

```diff
@@ -0,0 +1,242 @@
+```
+
+### [Cmd](#tab/cmd)
+
+```cmd
+az account show --query "{tenantId:tenantId,subscriptionid:id}"
+```
+
+---
+
+### Query examples for Azure Active Directory service principals
+
+This section shows example queries for AAD service principals.
+
+* The following query returns the first Microsoft Graph application service principal who has read permissions.
+
+### [Bash](#tab/bash)
+
+```azurecli-interactive
+az ad sp list --display-name "Microsoft Graph" --query '[0].appRoles[?value==`User.Read.All` && contains(allowedMemberTypes, `Application`)].id' --output tsv
+```
+
+### [PowerShell](#tab/powershell)
+
+```powershell-interactive
+az ad sp list --display-name "Microsoft Graph" --query "[0].appRoles[?value=='User.Read.All' && contains(allowedMemberTypes, 'Application')].id" --output tsv
+```
+
+### [Cmd](#tab/cmd)
+
+```cmd
+az ad sp list --display-name "Microsoft Graph" --query "[0].appRoles[?value=='User.Read.All' && contains(allowedMemberTypes, 'Application')].id" --output tsv
+```
+
+---
+
+### Query examples for storage accounts
+
+This section shows example queries for storage accounts.
+
+* This example returns the primary endpoints for all tables a storage account.
+
+### [Bash](#tab/bash)
+
+```azurecli-interactive
+az storage account show --resource-group QueryDemo --name mystorageaccount --query 'primaryEndpoints.table'
+```
+
+### [PowerShell](#tab/powershell)
+
+```powershell-interactive 
+az storage account show --resource-group QueryDemo --name mystorageaccount --query "primaryEndpoints.table"
+```
+
+### [Cmd](#tab/cmd)
+
+```cmd
+az storage account show --resource-group QueryDemo --name mystorageaccount --query "primaryEndpoints.table"
+```
+
+---
+
+### Query examples for Virtual Machines
+
+This section shows example queries for Virtual Machines (VMs).
+
+* This example returns the names of VMs whose disk size is larger than 50GB.
+
+### [Bash](#tab/bash)
+
+```azurecli-interactive
+az vm list --resource-group QueryDemo --query '[?storageProfile.osDisk.diskSizeGb >=`50`].{Name:name,  admin:osProfile.adminUsername, DiskSize:storageProfile.osDisk.diskSizeGb }' --output table
+```
+
+### [PowerShell](#tab/powershell)
+
+```powershell-interactive
+az vm list --resource-group QueryDemo --query "[?storageProfile.osDisk.diskSizeGb >=``50``].{Name:name,  admin:osProfile.adminUsername, DiskSize:storageProfile.osDisk.diskSizeGb }" --output table
+```
+
+Notice the extra escape characters (`\``) surrounding the 50 in the command above. These extra escape characters are present because Azure CLI commands are considered Command Prompt scripts, so both PowerShell and Command Prompt's parsing need to be taken into consideration. Azure CLI will only receive a symbol if it still exists after 2 rounds of parsing. For more information about other possible quoting issues please see [Quoting issues with PowerShell](https://github.com/Azure/azure-cli/blob/dev/doc/quoting-issues-with-powershell.md).
```

Comment:
```suggestion
Notice the extra escape characters (`` ` ``) surrounding the 50 in the command above. These extra escape characters are present because Azure CLI commands are considered Command Prompt scripts, so both PowerShell and Command Prompt's parsing need to be taken into consideration. Azure CLI will only receive a symbol if it still exists after 2 rounds of parsing. For more information about other possible quoting issues please see [Quoting issues with PowerShell](https://github.com/Azure/azure-cli/blob/dev/doc/quoting-issues-with-powershell.md).
```